types/node/index.d.ts

```diff
@@ -6818,3 +6819,205 @@ declare module "http2" {
+		/**
+		 * The total number of milliseconds elapsed for this entry.
+		 * This value will not be meaningful for all Performance Entry types.
+		 */
+		readonly duration?: number;
+
+		/**
+		 * The name of the performance entry.
+		 */
+		readonly name?: string;
+
+		/**
+		 * The high resolution millisecond timestamp marking the starting time of the Performance Entry.
+		 */
+		readonly startTime?: number;
+
+		/**
+		 * The type of the performance entry.
+		 * Currently it may be one of: 'node', 'mark', 'measure', 'gc', or 'function'.
+		 */
+		readonly entryType?: string;
+
+		/**
+		 * When performanceEntry.entryType is equal to 'gc', the performance.kind property identifies
+		 * the type of garbage collection operation that occurred.
+		 * The value may be one of:
+		 * - perf_hooks.constants.NODE_PERFORMANCE_GC_MAJOR
+		 * - perf_hooks.constants.NODE_PERFORMANCE_GC_MINOR
+		 * - perf_hooks.constants.NODE_PERFORMANCE_GC_INCREMENTAL
+		 * - perf_hooks.constants.NODE_PERFORMANCE_GC_WEAKCB
+		 */
+		readonly kind?: number;
+	}
+
+	interface PerformanceNodeTiming extends PerformanceEntry {
+		/**
+		 * The high resolution millisecond timestamp at which the Node.js process completed bootstrap.
+		 */
+		readonly bootstrapComplete?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which cluster processing ended.
+		 */
+		readonly clusterSetupEnd?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which cluster processing started.
+		 */
+		readonly clusterSetupStart?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which the Node.js event loop exited.
+		 */
+		readonly loopExit?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which the Node.js event loop started.
+		 */
+		readonly loopStart?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which main module load ended.
+		 */
+		readonly moduleLoadEnd?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which main module load started.
+		 */
+		readonly moduleLoadStart?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which the Node.js process was initialized.
+		 */
+		readonly nodeStart?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which preload module load ended.
+		 */
+		readonly preloadModuleLoadEnd?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which preload module load started.
+		 */
+		readonly preloadModuleLoadStart?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which third_party_main processing ended.
+		 */
+		readonly thirdPartyMainEnd?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which third_party_main processing started.
+		 */
+		readonly thirdPartyMainStart?: number;
+
+		/**
+		 * The high resolution millisecond timestamp at which the V8 platform was initialized.
+		 */
+		readonly v8Start?: number;
+    }
+
+	interface Performance {
```

Comment:
All the properties in `Performance` shouldn't be optional.
You mentioned the Async Hook Callbacks have been used as reference but this is not applicable as there user has to supply callbacks which are optional whereas these here are NodeJS APIs always present.